torchqdynamics/operators.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import annotations
+
+from math import prod
+
+import torch
+from torch import Tensor, device, dtype
+
+from .utils import _extract_tuple_from_varargs, tensprod
+
+__all__ = [
+    'sigmax',
+    'sigmay',
+    'sigmaz',
+    'sigmap',
+    'sigmam',
+    'qeye',
+    'destroy',
+    'create',
+    'displace',
+    'squeeze',
+]
+
+
+def sigmax(*, dtype: dtype = torch.complex128, device: device | None = None) -> Tensor:
+    """Pauli X operator."""
+    return torch.tensor([[0.0, 1.0], [1.0, 0.0]], dtype=dtype, device=device)
+
+
+def sigmay(*, dtype: dtype = torch.complex128, device: device | None = None) -> Tensor:
+    """Pauli Y operator."""
+    return torch.tensor([[0.0, -1.0j], [1.0j, 0.0]], dtype=dtype, device=device)
+
+
+def sigmaz(*, dtype: dtype = torch.complex128, device: device | None = None) -> Tensor:
+    """Pauli Z operator."""
+    return torch.tensor([[1.0, 0.0], [0.0, -1.0]], dtype=dtype, device=device)
+
+
+def sigmap(*, dtype: dtype = torch.complex128, device: device | None = None) -> Tensor:
+    """Pauli raising operator."""
+    return torch.tensor([[0.0, 1.0], [0.0, 0.0]], dtype=dtype, device=device)
+
+
+def sigmam(*, dtype: dtype = torch.complex128, device: device | None = None) -> Tensor:
+    """Pauli lowering operator."""
+    return torch.tensor([[0.0, 0.0], [1.0, 0.0]], dtype=dtype, device=device)
+
+
+def qeye(
+    *dims: int, dtype: dtype = torch.complex128, device: device | None = None
+) -> Tensor:
+    """Identity operator."""
+    dims = _extract_tuple_from_varargs(dims)
+    dim = prod(dims)
+    return torch.eye(dim, dtype=dtype, device=device)
```

Comment:
For example, after the proposed changes:
```suggestion
def qeye(*dims: int) -> Tensor:
    """Identity operator."""
    return torch.eye(prod(dims))
```

Reply:
Should we call this guy `eye` directly or `identity`? I never liked `qeye`.